services/frontend-service/pkg/cmd/server.go

```diff
@@ -314,9 +314,19 @@ func runServer(ctx context.Context) error {
 			When the user requests any path under "/ui", we always return the same index.html (because it's a single page application).
 			Anything else may be another valid rest request, like /health or /release.
 			*/
-			if strings.HasPrefix(req.URL.Path, "/ui") {
+			isUi := strings.HasPrefix(req.URL.Path, "/ui")
+			isHtml := req.URL.Path == "/" || req.URL.Path == "/index.html"
+			shouldCache := isUi || isHtml
+			if shouldCache {
+				resp.Header().Set("Cache-Control", "no-cache,no-store,must-revalidate,max-age=0")
```

Comment:
This can be a bit misleading as the variable name is `shouldCache` but we set a hearder called `no-cache` (might be a nitpick)

Reply:
done!